### DISCUSSION TITLE
What does `model.model.trend.trend_deltas` show?

### DISCUSSION THREAD

**Discussion by codeananda** [ourownstory/neural_prophet#1330] [Q&A - get help using NeuralProphet]
I want to find the difference between the slope/gradient of the trend between each of the changepoints (set by `n_changepoints`). Is there somewhere in the model that gives me this? 

I thought `model.model.trend.trend_deltas` might be the answer. But I'm confused. Is it really the change between trends? How is it calculated? It's defo not just a subtraction of one trend from the other.

What does the first number refer to?

```python
# Create toy dataset
start_values = [10, 15, 13, 13, -3, 12, 11]
end_values = [15, 13, 13, -3, 12, 11, 14]

num_points = 50
numbers = []
for start, end in zip(start_values, end_values):
    segment = np.linspace(start, end, num_points, endpoint=False)
    numbers.extend(segment)

# Format data correctly for NP
series = pd.Series(numbers)
ds = pd.date_range(start="2020-01-01", periods=len(series), freq="D")
y = series.values.reshape(-1)
full_df = pd.DataFrame({"ds": ds, "y": y})


model = NeuralProphet(n_changepoints=5)

set_random_seed(0)
model.fit(full_df)

forecast_df = model.predict(full_df, decompose=True, raw=False)

print(model.model.trend.trend_deltas)

fig, ax = plt.subplots(figsize=plt.figaspect(1/3))
ax.plot(forecast_df.y, label='Actual')
plt.plot(forecast_df.yhat1, label='Predicted')
ax.legend()
plt.show()

model.plot_components(forecast_df, plotting_backend='mat')
```

Output

```
Parameter containing:
tensor([[[ 1.4519, -0.8788,  0.3067, -7.0247,  3.5654,  0.7634]]],
       requires_grad=True)
```

<img width="637" alt="image" src="https://github.com/ourownstory/neural_prophet/assets/51246969/30f91ca1-57b9-4202-882c-46e9022acaef">

<img width="643" alt="image" src="https://github.com/ourownstory/neural_prophet/assets/51246969/9e28b0e3-d28a-4a0d-9917-0ea7d6c3f7ff">

@noxan @ourownstory
